spring-batch-samples/src/test/java/org/springframework/batch/sample/RemoteChunkingJobFunctionalTests.java

```diff
@@ -0,0 +1,182 @@
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.amqp.core.AmqpTemplate;
+import org.springframework.amqp.rabbit.connection.CachingConnectionFactory;
+import org.springframework.amqp.rabbit.connection.ConnectionFactory;
+import org.springframework.amqp.rabbit.core.RabbitTemplate;
+import org.springframework.batch.core.ExitStatus;
+import org.springframework.batch.core.Job;
+import org.springframework.batch.core.JobExecution;
+import org.springframework.batch.core.configuration.annotation.JobBuilderFactory;
+import org.springframework.batch.core.step.tasklet.TaskletStep;
+import org.springframework.batch.integration.chunk.RemoteChunkingMasterStepBuilderFactory;
+import org.springframework.batch.integration.config.annotation.EnableBatchIntegration;
+import org.springframework.batch.item.ItemReader;
+import org.springframework.batch.item.support.ListItemReader;
+import org.springframework.batch.sample.config.JobRunnerConfiguration;
+import org.springframework.batch.sample.remotechunking.WorkerConfiguration;
+import org.springframework.batch.test.JobLauncherTestUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.integration.amqp.dsl.Amqp;
+import org.springframework.integration.channel.DirectChannel;
+import org.springframework.integration.channel.QueueChannel;
+import org.springframework.integration.config.EnableIntegration;
+import org.springframework.integration.dsl.IntegrationFlow;
+import org.springframework.integration.dsl.IntegrationFlows;
+import org.springframework.messaging.PollableChannel;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import java.util.Arrays;
+
+/**
+ * <p>Ensure a RabbitMQ instance is running, modifying default.amqp.properties if needed.</p>
+ *
+ * <p>The master step of the tested job will read data and send chunks to the worker
+ * (started in {@link RemoteChunkingJobFunctionalTests#setUp()}) for processing and writing.</p>
+ */
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration(classes = {
+		JobRunnerConfiguration.class,
+		RemoteChunkingJobFunctionalTests.MasterConfiguration.class})
+public class RemoteChunkingJobFunctionalTests {
+
+	@Autowired
+	private JobLauncherTestUtils jobLauncherTestUtils;
+
+	private AnnotationConfigApplicationContext workerApplicationContext;
+
+	@Before
+	public void setUp() {
+		workerApplicationContext = new AnnotationConfigApplicationContext(WorkerConfiguration.class);
+		System.out.println("Started a worker..");
```

Comment:
Should remove system.outs...

Reply:
I thought that's fine for the sample (not production code). Removed in 83d43d60c578f475ab08d7262edee51cfce0d1a4